src/actions/sources.js

```diff
@@ -186,7 +188,11 @@ export function selectSource(id: string, options: SelectSourceOptions = {}) {
     let source = getSource(getState(), id);
 
     if (!source) {
-      return;
+      // If there is no source we deselect the current selected source
+      return dispatch({
+        type: constants.SELECT_SOURCE,
+        source: { id, url: null }
+      });
```

Comment:
Hmm, I think we should probably have a new action `clearSelectedSource`... Also, is the ID used later?